akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -434,12 +412,12 @@ class TLSActor(
     if (tracing) log.debug("handshake finished")
     val session = engine.getSession
 
-    hostInfo.map(_._1) match {
-      case Some(hostname) if !hostnameVerifier.verify(hostname, session) ⇒
```

Comment:
Yeah true, rather unlikely but we should properly deprecate/move away from the manual one perhaps.
I made a ticket for it: https://github.com/akka/akka/issues/21854